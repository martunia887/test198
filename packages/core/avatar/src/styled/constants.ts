import { gridSize } from '@atlaskit/theme';
import { SizeType, SupportedSizeWithAnIcon } from '../types';

export const TRANSITION_DURATION = '200ms';

const gridSizeValue: number = gridSize();

export const AVATAR_SIZES: SizeType = {
  xsmall: gridSizeValue * 2,
  small: gridSizeValue * 3,
  medium: gridSizeValue * 4,
  large: gridSizeValue * 5,
  xlarge: gridSizeValue * 12,
  xxlarge: gridSizeValue * 16,
};

// border radius only applies to "square" avatars
export const AVATAR_RADIUS: SizeType = {
  xsmall: 2,
  small: 2,
  medium: 3,
  large: 3,
  xlarge: 6,
  xxlarge: 12,
};

export const BORDER_WIDTH: SizeType = {
  xsmall: 2,
  small: 2,
  medium: 2,
  large: 2,
  xlarge: 2,
  xxlarge: 2,
};

export const EXCESS_INDICATOR_FONT_SIZE: SupportedSizeWithAnIcon = {
  small: 10,
  medium: 11,
  large: 12,
  xlarge: 16,
};

export const ICON_SIZES: SupportedSizeWithAnIcon = {
  small: 12,
  medium: 14,
  large: 15,
  xlarge: 18,
};

export const ICON_OFFSET: SupportedSizeWithAnIcon = {
  small: 0,
  medium: 0,
  large: 1,
  xlarge: 7,
};

export const SQUARE_ICON_OFFSET: SupportedSizeWithAnIcon = {
  small: 0,
  medium: 0,
  large: 0,
  xlarge: 1,
};
