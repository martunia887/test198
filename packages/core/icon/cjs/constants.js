"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sizes = {
    small: '16px',
    medium: '24px',
    large: '32px',
    xlarge: '48px',
    xxlarge: '96px',
};
exports.sizeMap = {
    small: 'small',
    medium: 'medium',
    large: 'large',
    xlarge: 'xlarge',
    xxlarge: 'xxlarge',
};
//# sourceMappingURL=constants.js.map