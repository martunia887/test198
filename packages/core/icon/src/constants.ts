import { sizeOpts } from './types';

export const sizes: { [key in sizeOpts]: string } = {
  small: '16px',
  medium: '24px',
  large: '32px',
  xlarge: '48px',
};
