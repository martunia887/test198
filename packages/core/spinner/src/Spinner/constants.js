// @flow
export const SIZES_MAP = {
  xsmall: 8,
  small: 16,
  medium: 24,
  large: 48,
  xlarge: 96,
};
export const DEFAULT_SIZE = SIZES_MAP.small;

const inAnimation = 1000;
const outAnimation = 530;
const outFadeDelay = 200;
const outFade = outAnimation - outFadeDelay;

export const durations: {
  inAnimation: number,
  outAnimation: number,
  outFadeDelay: number,
  outFade: number,
} = {
  inAnimation,
  outAnimation,
  outFadeDelay,
  outFade,
};

export const secondsDurations: {
  inAnimation: string,
  outAnimation: string,
  outFadeDelay: string,
  outFade: string,
  // $ExpectError - flow is very bad at tracing types through these transforms
} = Object.entries(durations).reduce(
  // $ExpectError - flow is very bad at tracing types through these transforms
  (acc, [key, milliseconds]) => ({ ...acc, [key]: `${milliseconds / 1000}s` }),
  {},
);
