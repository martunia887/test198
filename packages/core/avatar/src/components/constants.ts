import { ICON_SIZES } from '../styled/constants';
import { AvatarPropTypes } from '../types';

export const validIconSizes: string[] = Object.keys(ICON_SIZES);

export const propsOmittedFromClickData: Array<keyof AvatarPropTypes> = [
  'onBlur',
  'onClick',
  'onFocus',
  'onKeyDown',
  'onKeyUp',
  'onMouseDown',
  'onMouseEnter',
  'onMouseLeave',
  'onMouseUp',
];
