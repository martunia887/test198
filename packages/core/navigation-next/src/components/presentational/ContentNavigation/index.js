// @flow

import React, { Component, Fragment } from 'react';
import deepEqual from 'deep-equal';
import Transition from 'react-transition-group/Transition';
import { NavigationAnalyticsContext } from '@atlaskit/analytics-namespaced-context';

import { transitionDurationMs } from '../../../common/constants';
import { ContainerNavigation, ProductNavigation } from './primitives';
import type { ContentNavigationProps, ContentNavigationState } from './types';

const HiddenDiv = props => <div css={{ display: 'none' }} {...props} />;

const ToggleContent = ({
  isVisible,
  experimental_hideNavVisuallyOnCollapse: EXPERIMENTAL_HIDE_NAV_VISUALLY_ON_COLLAPSE,
  ...rest
}: {
  isVisible: boolean,
  experimental_hideNavVisuallyOnCollapse: boolean,
}) => {
  // If FF is false, retain the old behaviour of
  // un-mounting navigation components on collapse
  if (!EXPERIMENTAL_HIDE_NAV_VISUALLY_ON_COLLAPSE && !isVisible) {
    return null;
  }

  const ToggleWrapper = isVisible ? Fragment : HiddenDiv;

  return <ToggleWrapper {...rest} />;
};

export default class ContentNavigation extends Component<
  ContentNavigationProps,
  ContentNavigationState,
> {
  isMounted = false;

  state = {
    cachedContainer: null,
  };

  componentDidMount() {
    this.isMounted = true;
  }

  shouldComponentUpdate(nextProps: ContentNavigationProps) {
    const { props } = this;
    return !deepEqual(props, nextProps);
  }

  static getDerivedStateFromProps(
    { container }: ContentNavigationProps,
    state: ContentNavigationState,
  ) {
    if (container && container !== state.cachedContainer) {
      // We cache the most recent container component in state so that we can
      // render it while the container layer is transitioning out, which is
      // triggered by setting the container prop to null.
      return { ...state, cachedContainer: container };
    }
    return null;
  }

  render() {
    const {
      container,
      isVisible,
      product: Product,
      experimental_hideNavVisuallyOnCollapse: EXPERIMENTAL_HIDE_NAV_VISUALLY_ON_COLLAPSE,
    } = this.props;
    const { cachedContainer: CachedContainer } = this.state;

    const shouldRenderContainer = Boolean(container);
    const ContainerComponent = CachedContainer || Fragment;

    return (
      <Fragment>
        <ProductNavigation>
          <ToggleContent
            experimental_hideNavVisuallyOnCollapse={
              EXPERIMENTAL_HIDE_NAV_VISUALLY_ON_COLLAPSE
            }
            isVisible={isVisible}
          >
            <NavigationAnalyticsContext
              data={{ attributes: { navigationLayer: 'product' } }}
            >
              <Product />
            </NavigationAnalyticsContext>
          </ToggleContent>
        </ProductNavigation>
        <Transition
          in={shouldRenderContainer}
          timeout={this.isMounted ? transitionDurationMs : 0}
          mountOnEnter
          unmountOnExit
        >
          {state => (
            <ContainerNavigation
              isEntering={state === 'entering'}
              isExiting={state === 'exiting'}
            >
              <ToggleContent
                experimental_hideNavVisuallyOnCollapse={
                  EXPERIMENTAL_HIDE_NAV_VISUALLY_ON_COLLAPSE
                }
                isVisible={isVisible}
              >
                <NavigationAnalyticsContext
                  data={{ attributes: { navigationLayer: 'container' } }}
                >
                  <ContainerComponent />
                </NavigationAnalyticsContext>
              </ToggleContent>
            </ContainerNavigation>
          )}
        </Transition>
      </Fragment>
    );
  }
}
