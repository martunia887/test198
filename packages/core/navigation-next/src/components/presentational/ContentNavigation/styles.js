// @flow

import { colors } from '@atlaskit/theme';

import { CONTENT_NAV_WIDTH } from '../../../common/constants';

import type { ModeColors } from '../../../theme/types';

const baseStyles = {
  boxSizing: 'border-box',
  height: '100%',
  left: 0,
  minWidth: CONTENT_NAV_WIDTH,
  overflowX: 'hidden',
  position: 'absolute',
  top: 0,
  width: '100%',
};

export default ({ product, container }: ModeColors) => () => ({
  container: {
    ...baseStyles,
    backgroundColor: container.background.default,
    color: container.text.default,
  },
  product: {
    ...baseStyles,
    backgroundColor: product.background.default,
    color: product.text.default,
  },
});
