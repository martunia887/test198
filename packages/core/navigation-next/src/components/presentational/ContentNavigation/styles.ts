import { colors } from '@atlaskit/theme';

import { CONTENT_NAV_WIDTH } from '../../../common/constants';

import { ModeColors } from '../../../theme/ts-types';

const baseStyles = {
  boxSizing: 'border-box',
  height: '100%',
  left: 0,
  minWidth: CONTENT_NAV_WIDTH,
  overflowX: 'hidden',
  position: 'absolute',
  top: 0,
  width: '100%',
};

export default ({ product }: ModeColors) => () => ({
  container: {
    ...baseStyles,
    backgroundColor: colors.N20,
    color: colors.N500,
  },
  product: {
    ...baseStyles,
    backgroundColor: product.background.default,
    color: product.text.default,
  },
});
