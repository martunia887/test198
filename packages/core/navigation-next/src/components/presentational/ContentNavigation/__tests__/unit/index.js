// @flow
import React from 'react';
import { mount, shallow } from 'enzyme';

import ContentNavigation from '../../index';
import { transitionDurationMs } from '../../../../../common/constants';

const defaultProps = {
  isVisible: false,
  product: () => null,
  experimental_hideNavVisuallyOnCollapse: false,
};

describe('ContentNavigation', () => {
  it('should not trigger animations on first-page load', () => {
    const wrapper = shallow(<ContentNavigation {...defaultProps} />);

    expect(wrapper.find('Transition').props().timeout).toBe(0);

    wrapper.setProps({ isVisible: true });

    expect(wrapper.find('Transition').props().timeout).toBe(
      transitionDurationMs,
    );
  });

  it('should continue rendering the container view while the container layer animates out', async () => {
    const Container = () => null;
    const wrapper = mount(<ContentNavigation {...defaultProps} isVisible />);

    expect(wrapper.find(Container)).toHaveLength(0);
    wrapper.setProps({ container: Container });
    expect(wrapper.find(Container)).toHaveLength(1);
    wrapper.setProps({ container: undefined });
    // Should continue rendering the Container even though we've unset the prop
    expect(wrapper.find(Container)).toHaveLength(1);
  });

  it('should unmount the container and product when nav is collapsed', async () => {
    const Container = () => null;
    const Product = () => null;
    const wrapper = mount(
      <ContentNavigation
        {...defaultProps}
        product={Product}
        container={Container}
        isVisible
      />,
    );

    expect(wrapper.find(Product)).toHaveLength(1);
    expect(wrapper.find(Container)).toHaveLength(1);

    wrapper.setProps({ isVisible: false });

    expect(wrapper.find(Product)).toHaveLength(0);
    expect(wrapper.find(Container)).toHaveLength(0);
  });

  it('should visually hide the container view when nav is collapsed when "experimental_hideNavVisuallyOnCollapse" is passed', () => {
    const Container = () => null;
    const wrapper = mount(
      <ContentNavigation
        {...defaultProps}
        container={Container}
        isVisible
        experimental_hideNavVisuallyOnCollapse
      />,
    );
    expect(wrapper.find(Container).length).toBe(1);
    expect(
      wrapper
        .find(Container)
        .parent()
        .is('div'),
    ).toBeFalsy();

    wrapper.setProps({ isVisible: false });
    expect(wrapper.find(Container).length).toBe(1);
    expect(wrapper.find(Container).parent()).toMatchSnapshot();
  });

  it('should visually hide the product view when nav is collapsed when "experimental_hideNavVisuallyOnCollapse" is passed', () => {
    const Product = () => null;
    const wrapper = mount(
      <ContentNavigation
        product={Product}
        isVisible
        experimental_hideNavVisuallyOnCollapse
      />,
    );
    expect(wrapper.find(Product).length).toBe(1);

    wrapper.setProps({ isVisible: false });

    expect(wrapper.find(Product).length).toBe(1);
    expect(wrapper.find(Product).parent()).toMatchSnapshot();
  });
});
