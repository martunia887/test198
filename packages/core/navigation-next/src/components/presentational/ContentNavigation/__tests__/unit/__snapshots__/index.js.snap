// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentNavigation should visually hide the container view when nav is collapsed when "experimental_hideNavVisuallyOnCollapse" is passed 1`] = `
<AnalyticsContext
  data={
    Object {
      "navigationCtx": Object {
        "attributes": Object {
          "navigationLayer": "container",
        },
      },
    }
  }
/>
`;

exports[`ContentNavigation should visually hide the product view when nav is collapsed when "experimental_hideNavVisuallyOnCollapse" is passed 1`] = `
<AnalyticsContext
  data={
    Object {
      "navigationCtx": Object {
        "attributes": Object {
          "navigationLayer": "product",
        },
      },
    }
  }
/>
`;
